### PR TITLE
Database: implemented needed methods and other db-related stuff

### PR DESCRIPTION
**Pull request by LeanderK**
- updated database model to new schema
- updated operations to work on the new schema
- updated `db.sql` init-script to the new schema, placed it in the resources-folder
- `DatabaseManager` now reads the db.sql script to init the database
- `DatabaseManager` now used ConnectionPools, resulting in a better performance and automatic reconnects
- if the system-property `dropSchema` is set to `true`, `DatabaseManager` now drops the schema before initialising (just start the object-service with `java -jar JARFILE -DdropSchema=true`
- updated the `main`-class to use the `DatabaseManager`
- small improvements in the AbstractOperations class, resulting in some small API-Changes
- revised `NotificationRestOperations` to incorporate the changes in AbstractOperations
- adjusted `TemplateResource`and `NotificationResource` from the rest-package to the database-api changes
- implemented `PlatformOperations`
- implemented `WorkerBalanceOperations`
- implemented `WorkerOperations`
- implemented `TasksOperations`
- implemented `PlatformOperations`

### CODE REVIEW

**Comment by marcelhollerbach** [comment]
DENIED

**Comment by marcelhollerbach** [comment]
- ready to go in my eyes

**Comment by LeanderK** [comment]
right into your eyes?

**Comment by marcelhollerbach** [comment]
yep - into the left one

File: src/main/java/edu/kit/ipd/crowdcontrol/objectservice/rest/resources/NotificationResource.java [hunk 1]
```diff
@@ -56,7 +57,11 @@ public Notification get(Request request, Response response) {
      */
     public Notification put(Request request, Response response) {
         Notification notification = request.attribute("input");
-        notification = operations.create(notification);
+        try {
+            notification = operations.create(notification);
+        } catch (IllegalArgumentException e) {
+            throw new BadRequestException("All parameters must be set!");
```
Comment (kelunik) [inline src/main/java/edu/kit/ipd/crowdcontrol/objectservice/rest/resources/NotificationResource.java:63]: Wenn wir es einzeln prüfen würden, könnten wir auch eine Hilfe gehen, was genau fehlt.

File: src/main/resources/config.sample.properties [hunk 2]
```diff
@@ -1,4 +1,9 @@
 # Database
+#valid values are the name of the enums from: http://www.jooq.org/javadoc/3.7.x/org/jooq/SQLDialect.html
+database.dialect=MYSQL
+#configuration needed if no connection pool is provided
 database.url = jdbc:mysql://localhost:3306/crowdcontrol
 database.username = crowdcontrol
 database.password = ...
+#or if a connection pool is provided
```
Comment (kelunik) [inline src/main/resources/config.sample.properties:8]: How do these pools work? Maybe add some documentation link.

File: src/main/java/edu/kit/ipd/crowdcontrol/objectservice/rest/resources/NotificationResource.java (hunk 1 above)
Comment (LeanderK) [inline src/main/java/edu/kit/ipd/crowdcontrol/objectservice/rest/resources/NotificationResource.java:63]: you can copy the message of the illegal-argument exception

File: src/main/resources/config.sample.properties (hunk 2 above)
Comment (LeanderK) [inline src/main/resources/config.sample.properties:8]: no, thats standard java sql

Comment (kelunik) [inline src/main/resources/config.sample.properties:8]: The person configuring the software might not be used to Java / Java SQL.

Comment (marcelhollerbach) [inline src/main/resources/config.sample.properties:8]: then he should simply use database.{url,username,password} 

A link can get invalid, if you want to add docs for pools add them here in the wiki and add this link.

Comment (LeanderK) [inline src/main/resources/config.sample.properties:8]: i can write a bit more i guess, but to have the managed connection-pool set up you already have had to get used to the Java SQL ecosystem. The setup is not made for users. Maybe the most helpful line would be:
If you don't know what this is you probably haven't configured it anyway.

**Pull request merged**